src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/users.service');
+
+const get = (req, res) => {
+  const allExpenses = expensesService.getAllExpenses();
+  const { userId, categories, from, to } = req.query;
+  let filteredExpenses = {};
+
+  if (categories && userId) {
+    filteredExpenses = allExpenses.filter(item => item.category === categories);
+
+    return res.send(filteredExpenses);
+  };
+
+  if (userId) {
+    filteredExpenses = allExpenses.filter(
+      item => item.userId === Number(userId));
+
+    return res.send(filteredExpenses);
+  }
+
+  if (allExpenses.length === 0) {
+    return res.send([]);
+  }
+
+  if (from && to) {
+    filteredExpenses = allExpenses.filter(
+      item => item.spentAt > from && item.spentAt < to);
+
+    return res.send(filteredExpenses);
+  }
+
+  return res.send(allExpenses);
+};
+
+const getOne = (req, res) => {
+  const { id } = req.params;
+
+  const item = expensesService.getExpensesById(id);
+
+  if (!item) {
+    res.sendStatus(404);
```

Comment:
add message according to task requirement, fix all cases
![image](https://github.com/mate-academy/node_accounting-app-v2/assets/70009205/002c491a-1f93-4b6e-8cef-bc274b622a68)